### PR TITLE
Eagerly fetch data in hot-storage

### PR DESCRIPTION
**Pull request by jsign**
This PR:
- Have a [workaround](https://github.com/filecoin-project/lotus/issues/5361) to allow not relying on `IpfsOnlineMode=true`
- Allow a flag/env to configure a hard-timeout for stuck retrievals.
- Improve UX by masking behavior in which the miner claims that retrieval will be done by a worker miner and not the miner that made the deal.
- The `StageCid` method also pulls the data now from the IPFS network.

The main motivation for this change is: https://github.com/textileio/textile/pull/533

### CODE REVIEW

File: cmd/powd/main.go [hunk 1]
```diff
@@ -387,6 +389,7 @@ func setupFlags() error {
 	pflag.String("ffsminerselector", "reputation", "Miner selector to be used by FFS: 'sr2', 'reputation'.")
 	pflag.String("ffsminerselectorparams", "", "Miner selector configuration parameter, depends on --ffsminerselector.")
 	pflag.String("ffsminimumpiecesize", "67108864", "Minimum piece size in bytes allowed to be stored in Filecoin.")
+	pflag.Duration("ffsretrievalnexteventtimeout", time.Hour, "Maximum amount of time to wait for the next retrieval event before erroring it.")
```
[the remark this answers is not in the document]
Comment (asutula) [reply cmd/powd/main.go:392]: Yea, I've been using that lately, quite nice.

File: ffs/manager/manager.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ var (
 		Hot: ffs.HotConfig{
 			Enabled: false,
 			Ipfs: ffs.IpfsConfig{
-				AddTimeout: 480, // 8 min
+				AddTimeout: 15 * 60, // 15min
```
[the remark this answers is not in the document]
Comment (asutula) [reply ffs/manager/manager.go:36]: Worth updating everywhere in the db using the massive change tool?

**Review by asutula — APPROVED** [review]
Looks good!

File: ffs/manager/manager.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jsign) [reply ffs/manager/manager.go:36]: Yep, I have that as a task in my pre-deployment steps. Using `powcfg` to do it :)

**Pull request merged**